### ISSUE TITLE
AADSTS65005: Dynamic scope is invalid (AUTHORIZATION_CODE_NOT_EXCHANGED_FOR_TOKEN)

### ISSUE THREAD

**Issue by trbnb** [OfficeDev/Outlook-SDK-Android#3]
Trying to run the example in the README.md raises an Exception on `mAuthContext.acquireToken()`:

```
AADSTS65005: Dynamic scope is invalid: application 00000000-0000-0000-0000-000000000000 does not exist. Request ID: 857ad1f1-f3ec-41e9-8478-b260eb77f343, Timestamp: 03/01/2016 07:48:21
Trace ID: c66681c9-3bbb-4bf4-8566-5d962d02afdb
Correlation ID: 8a12c836-df4a-4eeb-b67d-36e84a7c7776
Timestamp: 2016-03-01 07:48:21Z
```

**Comment by trbnb** [comment]
It turns out this happens while getting the bearer token from "https://login.microsoftonline.com/common/oauth2/v2.0/token". It fails because the `scope` should be part of the request body.

**Comment by william8th** [comment]
If this is not already obvious, this project doesn't seem to be actively maintained anymore. I had reported another issue but there was no reply from any of the contributors. I'm also facing the same issue that you're mentioning here and I've found out that Microsoft has [changed their v2 Auth Protocol](https://blogs.technet.microsoft.com/ad/2016/01/27/for-developers-important-upcoming-changes-to-the-v2-0-auth-protocol/) and this SDK is not updated to work with the new protocol.

Either someone has to take over the development to patch the SDK or deal with Microsoft's server APIs from scratch.

**Comment by trbnb** [comment]
Yes, this is the impression that I had as well but I wanted to try to contact someone anyway. 

Just to let you know: I think the problem lies in the `OAuth2.java` file from the Azure AD library for Android that is used in this libraries `AuthenticationContext`: https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/master/src/src/com/microsoft/aad/adal/Oauth2.java#L153

As you can see they leave out the `scope` parameter when building the Token-request body.

But if this library isn't actively maintained anymore I just build my own.
